tests/test_graph_components/providers/test_domain_without_responses_provider.py

```diff
@@ -0,0 +1,60 @@
+from typing import Text
+
+import pytest
+from rasa.graph_components.providers.domain_without_response_provider import (
+    DomainWithoutResponseProvider,
+)
+from rasa.engine.graph import ExecutionContext
+from rasa.engine.storage.resource import Resource
+from rasa.engine.storage.storage import ModelStorage
+from rasa.shared.core.domain import KEY_E2E_ACTIONS, KEY_RESPONSES, Domain
+
+
+@pytest.mark.parametrize(
+    "domain_yml",
+    [
+        ("data/test_domains/travel_form.yml"),
+        ("data/test_domains/mixed_retrieval_intents.yml"),
+    ],
+)
+def test_copy_without_responses(
+    default_model_storage: ModelStorage,
+    default_execution_context: ExecutionContext,
+    domain_yml: Text,
+):
+
+    component = DomainWithoutResponseProvider.create(
+        {"arbitrary-unused": 234},
+        default_model_storage,
+        Resource("xy"),
+        default_execution_context,
+    )
+
+    domain = Domain.from_file(path=domain_yml,)
+    domain_copy = component.recreate_without_responses(domain=domain,)
+
+    given_as_dict = domain.as_dict()
+    copy_as_dict = domain_copy.as_dict()
+
+    # all configurations not impacted by responses stay intact
+    assert sorted(given_as_dict.keys()) == sorted(copy_as_dict.keys())
+    for key in given_as_dict.keys():
+        if key not in [KEY_RESPONSES, KEY_E2E_ACTIONS]:
+            assert given_as_dict[key] == copy_as_dict[key]
+
+    # Note: The given domains do contain responses and some actions for which no
+    # responses are defined...
+    assert domain.responses, "choose a different test config"
+    assert domain.user_actions, "choose a different test config"
```

Comment:
I'm not entirely sure what role the `choose a different test config` string plays in the assertion? 🤔

Reply:
Just in case someone decides to adapt the .ymls we have in the test_domain folder and changes them so that they don't have the nice properties that I would like to use for testing here - then these asserts will function as a reminder that these tests have become outdated

Reply:
I've made this more verbose/explicit and added a bit on how to resolve it